test/python/pulse/test_pulse_lib.py

```diff
@@ -121,6 +121,24 @@ def test_construction(self):
         Constant(duration=150, amp=0.1 + 0.4j)
         Drag(duration=25, amp=0.2 + 0.3j, sigma=7.8, beta=4)
 
+    # This test should be removed once deprecation of complex amp is completed.
+    def test_complex_amp_deprecation(self):
+        """Test that deprecation warnings and errors are raised for complex amp"""
+        with self.assertWarns(PendingDeprecationWarning):
```

Comment:
You are doing things correctly. You can have immediate `PendingDeprecationWarning` according to the [deprecation policy](https://qiskit.org/documentation/deprecation_policy.html#removing-a-feature).